lib/_stream_writable.js

```diff
@@ -134,11 +134,28 @@ Object.defineProperty(WritableState.prototype, 'buffer', {
      'instead.')
 });
 
+// Test _writableState for inheritance to account for Duplex streams,
+// whose prototype chain only points to Readable.
+var realHasInstance;
+if (Symbol.hasInstance) {
```

Comment:
We should feature-detect that `Symbol` exists to avoid a regexp in `readable-stream`. Some of our supported platforms there do not have symbols yet.